### PR TITLE
Update proxy.py to strip the main function out of an optional struct

### PR DESCRIPTION
**Pull request by jberstler**
…envelope

OpenWhisk actions are expected to declare a global "main" function
However, this does not work well if you want to develop multiple actions in a single Xcode target, as you cannot re-declare a global main function.
This solution suggests that users working in Xcode may optionally declare their main function inside a struct, in order to solve the namespacing issue at compliation time.
Just before the .swift code is run, if a struct declaration is found, peel away the struct to reveal the delicious main function contained inside.

### CODE REVIEW

**Comment by jberstler** [comment]
I'm changing an issue comment.

File: core/swiftAction/proxy.py [hunk 1]
```diff
@@ -43,8 +43,50 @@ def init():
     message = message.get("value", {})
 
     if "code" in message:
+        # if the main function is wrapped in a struct, this code will strip the
+        # struct declaration away, freeing the main function. If not, the result
+        # will be the same as code
+        codeWithoutStruct = ""
```
Comment (bjustin-ibm) [inline core/swiftAction/proxy.py]: This is a comment about the code.

**Comment by jberstler** [comment]
I'm adding a second comment.

File: core/swiftAction/proxy.py (hunk 1 above)
Comment (jberstler) [inline core/swiftAction/proxy.py:49]: Adding a review comment.

Comment (jberstler) [inline core/swiftAction/proxy.py:49]: Editing a reply to a review comment. And again.

**Comment by jberstler** [comment]
Adding another new comment.

File: core/swiftAction/proxy.py (hunk 1 above)
Comment (jberstler) [inline core/swiftAction/proxy.py:49]: Add another review comment.

Comment (jberstler) [inline core/swiftAction/proxy.py:49]: Adding a review comment to a ready PR.

**Comment by jberstler** [comment]
Adding a new comment to a ready PR.

**Comment by bjustin-ibm** [comment]
new round of comment testing

File: core/swiftAction/proxy.py (hunk 1 above)
Comment (bjustin-ibm) [inline core/swiftAction/proxy.py]: A new round of review comments

Comment (bjustin-ibm) [inline core/swiftAction/proxy.py]: Let's try adding a review comment to a PR in review.

Comment (jberstler) [inline core/swiftAction/proxy.py:49]: Add a new review comment.

**Comment by jberstler** [comment]
MOAR COMMENTS!!!

**Comment by jberstler** [comment]
This shouldn't trigger a change because the PR is ready.

**Pull request closed without merging**